Add a note in migration guide about the custom scalar compat mode (#3488)
apollographql/apollo-android: docs/source/migration/3.0.mdx

This page describes the most important changes, along with how to migrate an existing project from Apollo Android 2.x to Apollo Android 3.x.

## The quick route 🚀

> Apollo Android 3.0 provides a few shortcuts and compatibility modes to ease the migration from 2.x.  To quickly arrive at a working state, follow these steps:

Reviewer: ```suggestion
Apollo Android 3 provides a few helpers and compatibility modes to ease the migration from 2.x. To quickly reach a working state, follow the steps below.
Once you have a working app, we strongly advise to read the rest of this document as the helpers will be removed in a future version of Apollo Android.
```

Reviewer: Also, we'll need to upgrade that text once we remove the helper is out to say something like "use 3.0 first and then migrate to 3.1"

1. Update your dependencies and imports (`com.apollographql.apollo` → `com.apollographql.apollo3`, see [section below](#package-name--group-id--plugin-id)).  
  Remove `apollo-coroutines-support` and `apollo-android-support` if applicable.
2. Gradle configuration:
```kotlin
apollo {
  // Remove this
  generateKotlinModels.set(true)
  
  // Add this

>   useVersion2Compat("com.example")

Reviewer: ```suggestion
  useVersion2Compat()
```

The argument is `rootPackageName`, which most users will not have set  I believe (I know this is confusing, this is why we're making packageName mandatory in 3.x)

}
```
3. Apollo Client configuration:
```kotlin
val client = ApolloClient.builder()
  .serverUrl(...)

  // Replace:
  .addCustomTypeAdapter(CustomType.YOURTYPE, ...)

  // With:
  .addCustomTypeAdapter(YourType.type, ...)

  .build()

```
4. Queries and other operations:
```kotlin
// Replace:
val response = apolloClient
  .query(MyQuery())
  .await()

// With:
val response = apolloClient
  .query(MyQuery())
  .execute()
```

> ## All changes

Reviewer: ```suggestion
## All details
```
